src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -275,10 +277,13 @@ private List<String> scriptLauncherCmd(ShellController c, FilePath ws, @CheckFor
 
         /** Caching zOS flag to avoid round trip calls in exitStatus()         */
         private final boolean isZos;
+        /** Encoding of jenkins-result.txt if on z/OS, null otherwise          */
+        private String jenkinsResultTxtEncoding;
 
-        private ShellController(FilePath ws, boolean zOsFlag, @NonNull String cookieValue) throws IOException, InterruptedException {
+        private ShellController(FilePath ws, boolean zOsFlag, @NonNull String cookieValue, String jenkinsResultTxtEncoding) throws IOException, InterruptedException {
             super(ws, cookieValue);
             this.isZos = zOsFlag;
+            this.jenkinsResultTxtEncoding = jenkinsResultTxtEncoding == null ? getCharset() : jenkinsResultTxtEncoding;
```

Comment:
This might cause problems on non-z/OS if `exitStatus` is called after a controller restart. Also probably incompatible after a restart in which this plugin is upgraded, where the new field will be deserialized to `null`. Just use
```suggestion
            this.jenkinsResultTxtEncoding = jenkinsResultTxtEncoding;
```